plugins/module_utils/redfish_utils.py

```diff
@@ -2273,6 +2277,14 @@ def virtual_media_insert(self, options):
         data = response['data']
         if 'VirtualMedia' not in data:
             return {'ret': False, 'msg': "VirtualMedia resource not found"}
+        if data["FirmwareVersion"].startswith("iLO 4"):
+            image_only = True
+            for k, v in options.items():
```

Comment:
Is there a strong need to proactively fail here for this? Like other properties, I would prefer this to be rejected by the service so we can minimize anything vendor-specific.

Reply:
The only issue I have with that is that the error message returned by Ansible is very vague as to what the problem is compared to the full error message returned by the API.

All you would see from Ansible is:
```
TASK [provision_host : community.general.redfish_command] ***************************************************************************************
fatal: [redacted -> localhost]: FAILED! => {"changed": false, "msg": "HTTP Error 400 on PATCH request to 'https://10.111.250.26/redfish/v1/Managers/1/VirtualMedia/2/', extended message: 'Bad Request'"}
```

Reply:
I could edit the _get_extended_message function and add an else to the try where it puts a string of the entire @Message.ExtendedInfo?  It looks like iLO 4 isn't quite sending what is expected.

Reply:
That might be a good idea. I'd like to avoid starting a pattern where we might proactively check the service's version on any modification operation, especially if the properties are being called out by the user. That could be an endless task.

Reply:
I have addressed this in the latest commits.